src/evict/evict_page.c

```diff
@@ -84,6 +84,89 @@ __wt_page_release_evict(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags)
     return (ret);
 }
 
+#define WT_EVICT_STATS_CLEAN 0x01
+#define WT_EVICT_STATS_FORCE_HS 0x02
+#define WT_EVICT_STATS_SUCCESS 0x04
+#define WT_EVICT_STATS_URGENT 0x08
+
+/*
+ * __evict_stats_update --
+ *     Update the stats of eviction.
+ *
+ */
+static void
+__evict_stats_update(WT_SESSION_IMPL *session, uint8_t flags)
+{
+    WT_CONNECTION_IMPL *conn;
+    uint64_t eviction_time, eviction_time_milliseconds;
+
+    conn = S2C(session);
+
+    if (session->evict_timeline.reentry_hs_eviction) {
+        session->evict_timeline.reentry_hs_evict_finish = __wt_clock(session);
+        eviction_time = WT_CLOCKDIFF_US(session->evict_timeline.reentry_hs_evict_finish,
+          session->evict_timeline.reentry_hs_evict_start);
+    } else {
+        session->evict_timeline.evict_finish = __wt_clock(session);
+        eviction_time = WT_CLOCKDIFF_US(
+          session->evict_timeline.evict_finish, session->evict_timeline.evict_start);
+    }
+    if (LF_ISSET(WT_EVICT_STATS_SUCCESS)) {
+        if (LF_ISSET(WT_EVICT_STATS_URGENT)) {
+            if (LF_ISSET(WT_EVICT_STATS_FORCE_HS))
+                WT_STAT_CONN_INCR(session, cache_eviction_force_hs_success);
+            if (LF_ISSET(WT_EVICT_STATS_CLEAN)) {
+                WT_STAT_CONN_INCR(session, cache_eviction_force_clean);
+                WT_STAT_CONN_INCRV(session, cache_eviction_force_clean_time, eviction_time);
+            } else {
+                WT_STAT_CONN_INCR(session, cache_eviction_force_dirty);
+                WT_STAT_CONN_INCRV(session, cache_eviction_force_dirty_time, eviction_time);
+            }
+        }
+
+        if (LF_ISSET(WT_EVICT_STATS_CLEAN))
+            WT_STAT_CONN_DATA_INCR(session, cache_eviction_clean);
+        else
+            WT_STAT_CONN_DATA_INCR(session, cache_eviction_dirty);
+
+        /* Count page evictions in parallel with checkpoint. */
+        if (conn->txn_global.checkpoint_running)
+            WT_STAT_CONN_INCR(session, cache_eviction_pages_in_parallel_with_checkpoint);
+    } else {
+        if (LF_ISSET(WT_EVICT_CALL_URGENT)) {
+            if (LF_ISSET(WT_EVICT_STATS_FORCE_HS))
+                WT_STAT_CONN_INCR(session, cache_eviction_force_hs_fail);
+            WT_STAT_CONN_INCR(session, cache_eviction_force_fail);
+            WT_STAT_CONN_INCRV(session, cache_eviction_force_fail_time, eviction_time);
+        }
+
+        WT_STAT_CONN_DATA_INCR(session, cache_eviction_fail);
+    }
+    if (!session->evict_timeline.reentry_hs_eviction) {
+        eviction_time_milliseconds = eviction_time / WT_THOUSAND;
+        if (eviction_time_milliseconds > conn->cache->evict_max_ms)
+            conn->cache->evict_max_ms = eviction_time_milliseconds;
+        if (eviction_time_milliseconds > WT_MINUTE * WT_THOUSAND)
+            __wt_verbose(session, WT_VERB_EVICT,
```

Comment:
It's right to change this `__wt_verbose_warn` to `__wt_verbose` (since 4.4 doesn't have this), but we should add a "Warning: " prefix to the line.

Reply:
easy, [88a8cc8](https://github.com/wiredtiger/wiredtiger/pull/9341/commits/88a8cc8f3e2a65dbfdf39f85751ffc88d44737f1)